packages/arb-ts/src/lib/assetBridger/tokenBridger.ts

```diff
@@ -0,0 +1,719 @@
+    } catch (err) {
+      if (
+        err instanceof Error &&
+        isError(err) &&
+        err.code === Logger.errors.CALL_EXCEPTION
+      ) {
+        return false
+      } else {
+        throw err
+      }
+    }
+  }
+
+  /**
+   * Is this a known or unknown WETH gateway
+   * @param gatewayAddress
+   * @param l1Provider
+   * @returns
+   */
+  private async isWethGateway(
+    gatewayAddress: string,
+    l1Provider: Provider
+  ): Promise<boolean> {
+    const wethAddress = this.l2Network.tokenBridge.l1WethGateway
+    if (this.l2Network.isCustom) {
+      // For custom network, we do an ad-hoc check to see if it's a WETH gateway
+      if (await this.isUnkownWethGateway(gatewayAddress, l1Provider)) {
+        return true
+      }
+      // ...otherwise we directly check it against the config file
+    } else if (wethAddress === gatewayAddress) {
+      return true
+    }
+    return false
+  }
+
+  /**
+   * Get the L2 token contract at the provided address
+   * @param l1Provider
+   * @param l1TokenAddr
+   * @returns
+   */
+  public getL2TokenContract(
+    l2Provider: Provider,
+    l2TokenAddr: string
+  ): StandardArbERC20 {
+    return StandardArbERC20__factory.connect(l2TokenAddr, l2Provider)
+  }
+
+  /**
+   * Get the L1 token contract at the provided address
+   * @param l1Provider
+   * @param l1TokenAddr
+   * @returns
+   */
+  public getL1TokenContract(l1Provider: Provider, l1TokenAddr: string): ERC20 {
+    return ERC20__factory.connect(l1TokenAddr, l1Provider)
+  }
+
+  /**
+   * Get the corresponding L2 for the provided L1 token
+   * @param erc20L1Address
+   * @param l1Provider
+   * @returns
+   */
+  public getL2ERC20Address(
+    erc20L1Address: string,
+    l1Provider: Provider
+  ): Promise<string> {
+    const l1GatewayRouter = L1GatewayRouter__factory.connect(
+      this.l2Network.tokenBridge.l1GatewayRouter,
+      l1Provider
+    )
+
+    return l1GatewayRouter.functions
+      .calculateL2TokenAddress(erc20L1Address)
+      .then(([res]) => res)
+  }
+
+  /**
+   * Get the corresponding L1 for the provided L2 token
+   * @param erc20L1Address
+   * @param l1Provider
+   * @returns
+   */
+  public getL1ERC20Address(
+    erc20L2Address: string,
+    l2Provider: Provider
+  ): Promise<string> {
+    const arbERC20 = StandardArbERC20__factory.connect(
+      erc20L2Address,
+      l2Provider
+    )
+    return arbERC20.functions.l1Address().then(([res]) => res)
+  }
+
+  /**
+   * Whether the token has been disabled on the router
+   * @param l1TokenAddress
+   * @param l1Provider
+   * @returns
+   */
+  public async getL1TokenIsDisabled(
+    l1TokenAddress: string,
+    l1Provider: Provider
+  ): Promise<boolean> {
+    const l1GatewayRouter = L1GatewayRouter__factory.connect(
+      this.l2Network.tokenBridge.l1GatewayRouter,
+      l1Provider
+    )
+
+    return (
+      (await l1GatewayRouter.l1TokenToGateway(l1TokenAddress)) ===
+      '0x0000000000000000000000000000000000000001'
```

Comment:
we could change precompile_addresses.ts to something like constants.ts and move this and the alias offset there, so it serves as a bit of a reference/spec

Reply:
added a constants file and moved the address offset + this